pandas/io/parsers/readers.py

```diff
@@ -1846,7 +1853,40 @@ def read(self, nrows: int | None = None) -> DataFrame:
             else:
                 new_rows = len(index)
 
-            df = DataFrame(col_dict, columns=columns, index=index)
+            if hasattr(self, "orig_options"):
```

Comment:
can we do something more explicit than a hasattr check?

Reply:
no, you can subclass the reader, so we don't have any control over it

Reply:
does anybody actually do this?  i judge those people, their ethics, and their hygiene.

Reply:
That's something I can't answer, we might want to deprecate maybe, but we are stuck with hasattr here until then